source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/observations.py

```diff
@@ -231,6 +237,92 @@ def image(
+        initialize_all: bool = False,
+        device: str = "cuda:0",
+    ):
+        super().__init__(cfg, env)
+        if model_zoo_cfg is None:
+            self.model_zoo_cfg = {
+                # For more info about Theia, see: https://theia.theaiinstitute.com/
+                "TheiaTiny": {
+                    "model": (
+                        lambda: AutoModel.from_pretrained(
+                            "theaiinstitute/theia-tiny-patch16-224-cddsv", trust_remote_code=True
+                        )
+                        .eval()
+                        .to(device)
+                    ),
+                    "preprocess": lambda img: (img - torch.amin(img, dim=(1, 2), keepdim=True)) / (
+                        torch.amax(img, dim=(1, 2), keepdim=True) - torch.amin(img, dim=(1, 2), keepdim=True)
+                    ),  # Rescale to [0, 1]
+                    "inference": lambda model, images: model.forward_feature(
+                        images, do_rescale=False, interpolate_pos_encoding=True
+                    ),
+                },
+                "ResNet18": {
+                    "model": lambda: models.resnet18(pretrained=True).eval().to(device),
+                    "preprocess": lambda img: (
+                        img.permute(0, 3, 1, 2)  # Convert [batch, height, width, 3] -> [batch, 3, height, width]
+                        # Normalize in the format expected by pytorch: https://pytorch.org/hub/pytorch_vision_resnet/
+                        - torch.tensor([0.485, 0.456, 0.406], device=img.device).view(1, 3, 1, 1)
+                    ) / torch.tensor([0.229, 0.224, 0.225], device=img.device).view(1, 3, 1, 1),
+                    "inference": lambda model, images: model(images),
+                },
+            }
+        self.reset_model(initialize_all=initialize_all)
+
+    # The following is named reset_model instead of reset as otherwise, it's called at the end of every episode
+    def reset_model(self, model_name: str | None = None, initialize_all: bool = False) -> None:
+        if model_name is None:
+            print("[WARNING]: No model name supplied, emptying entire model zoo.")
+            self.model_zoo = {}
+        elif model_name is not None:
+            self.model_zoo[model_name] = self.model_zoo_cfg[model_name]["model"]()
+        if initialize_all:
+            for model_name, model_callables in self.model_zoo_cfg.items():
+                self.model_zoo[model_name] = model_callables["model"]()
+
+    def __call__(
+        self,
+        env: ManagerBasedEnv,
+        sensor_cfg: SceneEntityCfg = SceneEntityCfg("tiled_camera"),
+        data_type: str = "rgb",
+        convert_perspective_to_orthogonal: bool = False,
+        model_name: str = "ResNet18",
+    ) -> torch.Tensor:
+        if model_name not in self.model_zoo:
+            print(f"[INFO]: Adding {model_name} to the model zoo")
+            self.model_zoo[model_name] = self.model_zoo_cfg[model_name]["model"]()
```

Comment:
A bit confused. Why not resolve params inside the constructor?

Reply:
Allows a sort of "lazy eval" where you can load a desired subset of a model zoo cfg into GPU memory.

Reply:
Constructor only consumes self, cfg: ObservationTermCfg, env: ManagerBasedEnv 

want to be able to load a subset of the entire model zoo for efficiency, this allows what subset to be configured